tests/unittests/Foundation/NSStringTests.m

```diff
@@ -607,4 +607,59 @@ void testUrlCharacterSetEncoding(NSString* decodedString, NSString* encodedStrin
 
     ASSERT_OBJCEQ(@"value1", propertyList[@"key1"]);
     ASSERT_OBJCEQ(@"value2", propertyList[@"key2"]);
-}
\ No newline at end of file
+}
+
+TEST(NSString, LastPathComponent) {
+    NSString* string = @"";
+    EXPECT_OBJCEQ(@"", string.lastPathComponent);
+
+    string = @"hello.world";
+    EXPECT_OBJCEQ(@"hello.world", string.lastPathComponent);
+
+    string = @"/";
+    EXPECT_OBJCEQ(@"/", string.lastPathComponent);
+
+#if TARGET_OS_WIN32
```

Comment:
nit: comment here about why this is Windows-only (I get that it's because of the drive char, but the comment still helps)